integration/substrate/arrays.sol

```diff
@@ -13,9 +13,7 @@ contract arrays {
 		Permission[] perms;
 	}
 
-	// declare a sparse array. Sparse arrays are arrays which are too large to
-	// fit into account data on Solana; this is not neccessarily a Solidity feature
-	user[type(uint64).max] users;
+	user[1024] users;
```

Comment:
This particular one is actually, the JS API does not support array dims > 2048

```
 1) Deploy arrays contract and test
       arrays in account storage:
     Error: PortableRegistry: 7: Error extracting {"path":[],"params":[],"def":{"array":{"len":100000024,"type":6}},"docs":[]}: Only support for [Type; <length>], where length <= 2048
```

Besides that, dealing with large arrays like that wouldn't be economic anyways?

Reply:
Why would it not be economic? This is a storage array, it's never allocated in memory. Access is per-element (each element of the array uses a storage slot).

Having said that, I don't know what to do about the javascript limitation. You could skip storage elements if they are too large.

Reply:
Sorry I misread, I thought in the test it would actually return the entire `users` array, not just a single one :facepalm: 

Hmm yes but that is a problem I'd rather solve later on, e.g. when a user actually needs this...